CONTRIBUTING.md

```diff
@@ -0,0 +1,292 @@
+
+After you fork the repo, [clone][clone-a-repo] your fork in your local workspace:
+
+```bash
+$ git clone git@github.com<YOUR-USERNAME>/utam-java.git
+$ cd utam-java
+```
+
+### 2) Install dependencies
+
+_We use [Maven](https://maven.apache.org/) for dependency management._
+
+### 2) Building UTAM Java
+
+```bash
+$ mvn package
+```
+
+This command does several things:
+
+1. It updates the dependencies of the project, downloading from the Maven Central repository,
+if necessary.
+2. It compiles the Java code of the project.
+3. It runs the unit tests defined by the project.
+4. It packages the compiled code into distributable artifacts (.jars).
+
+The Maven project file will build two artifacts:
+
+- `utam-compiler`: the compiler used to create Java source code from UTAM Page Object declarative
+description files (*.utam.json)
+- `utam-core`: the core runtime library used by tests to consume UTAM Page Objects
+
+## Testing
+
+### Unit Testing UTAM
+
+We use [TestNG](https://testng.org) to write unit tests for UTAM.
+
+Run unit tests by executing the following command from the repository's root:
+
+```bash
+$ mvn test
+```
+
+Tests can also be run and debugged using the TestNG integraation plugin for a number of Java-aware
+IDEs. UTAM developers usually use [IntelliJ IDEA](https://www.jetbrains.com/idea/) or [Eclipse](https://www.eclipse.org/ide/).
+
+## Editor Configurations
+
+Configure your editor to use our lint and code style rules to speed up the code review process!
+
+### Code Style
+
+UTAM Java uses the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).
+Format settings definition files for importing into IDEs are available for
+[Eclipse](https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml)
+and [IntelliJ IDEA](https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml).
+
+## Git Workflow
+
+The process of submitting a pull request is straightforward and generally follows the same pattern each time:
+
+1. [Fork the UTAM Java repo](#fork-the-utam-java-repo)
+2. [Create a feature branch](#create-a-feature-branch)
+3. [Make your changes](#make-your-changes)
+4. [Rebase](#rebase)
+5. [Check your submission](#check-your-submission)
+6. [Create a pull request](#create-a-pull-request)
+7. [Update the pull request](#update-the-pull-request)
+8. [Commit Message Guidelines](#commit-message-conventions)
+
+### Fork the UTAM Java repo
+
+[Fork][fork-a-repo] the [salesforce/utam-java](https://github.com/salesforce/utam-java) repo. Clone your fork in your local
+workspace and [configure][configuring-a-remote-for-a-fork] your remote repository settings.
+
+```bash
+$ git clone git@github.com:<YOUR-USERNAME>/utam-java.git
+$ cd utam-java
+$ git remote add upstream git@github.com:salesforce/utam-java.git
+```
+
+### Create a feature branch
+
+```bash
+$ git checkout main
```

Comment:
of main? maybe dev or for external contributors it's main?

Reply:
fixed